src/datajudge/constraints/numeric.py

```diff
@@ -158,3 +158,79 @@ def test(self, engine: sa.engine.Engine) -> TestResult:
         )
         result = deviation <= self.max_absolute_deviation
         return TestResult(result, assertion_text)
+
+
+class NumericPercentile(Constraint):
+    def __init__(
+        self,
+        ref: DataReference,
+        k: float,
```

Comment:
Minor nitpicking, I never liked the name just being `k` in here. Now, when reading further down some docstring where it mentions the `k-th` percentile, it made sense, but it lacks context here. The main reason is that we decided not to document in docstrings here but in the convenience functions.

Reply:
I first couldn't come up with a better technical term for it. But I actually think we could just call it `percentage`. :)

Reply:
Or perhaps just `percentile`?

Reply:
I think that wouldn't be what we want. The way I understand the terminology it holds that:
$\Pr[x <= percentile] = percentage / 100$

where there is incongruency wrt the strictness of the inequality.